packages/rich-text/src/component/use-anchor.js

```diff
@@ -174,7 +148,35 @@ export function useAnchor( { editableContentElement, settings = {} } ) {
 			editableContentElement.removeEventListener( 'focusin', attach );
 			editableContentElement.removeEventListener( 'focusout', detach );
 		};
-	}, [ editableContentElement, tagName, className ] );
+	}, [ editableContentElement, tagName, className, callback ] );
+
+	return {
+		anchor,
+		update: callback,
+	};
+}
+
+/**
+ * This hook, to be used in a format type's Edit component, returns the active
+ * element that is formatted, or a virtual element for the selection range if
+ * no format is active. The returned value is meant to be used for positioning
+ * UI, e.g. by passing it to the `Popover` component via the `anchor` prop.
+ *
+ * @param {Object}           $1                        Named parameters.
+ * @param {HTMLElement|null} $1.editableContentElement The element containing
+ *                                                     the editable content.
+ * @param {WPFormat=}        $1.settings               The format type's settings.
+ * @return {Element|VirtualAnchorElement|undefined|null} The active element or selection range.
+ */
+export function useAnchor( { editableContentElement, settings = {} } ) {
+	const { anchor } = useAnchorBase( { editableContentElement, settings } );
```

Comment:
We cannot modify the existing public API of the original `useAnchor` to expose an `update` method because it only returns a single object. That object can be interogated by consumers and thus it is not safe to expose additional properties on it (trust me I tried and it immediately broke the autocompleter tests).

The hook continues to behave exactly as it did before, except that the implementation has moved to the "base" hook and we simply call that and expose only the `anchor` from the `useAnchor` hook.